src/imp/unix.rs

```diff
@@ -99,24 +114,30 @@ pub fn reopen(file: &File, path: &Path) -> io::Result<File> {
     }
 }
 
+#[cfg(not(target_os = "redox"))]
 pub fn persist(old_path: &Path, new_path: &Path, overwrite: bool) -> io::Result<()> {
     unsafe {
         let old_path = try!(cstr(old_path));
         let new_path = try!(cstr(new_path));
         if overwrite {
-            if rename(old_path.as_ptr() as *const c_char,
-                      new_path.as_ptr() as *const c_char) != 0 {
-                return Err(io::Error::last_os_error());
-            }
+            try!(cvt_err(rename(old_path.as_ptr() as *const c_char,
+                                new_path.as_ptr() as *const c_char)));
         } else {
-            if link(old_path.as_ptr() as *const c_char,
-                    new_path.as_ptr() as *const c_char) != 0 {
-                return Err(io::Error::last_os_error());
-            }
+            try!(cvt_err(link(old_path.as_ptr() as *const c_char,
+                              new_path.as_ptr() as *const c_char)));
             // Ignore unlink errors. Can we do better?
             // On recent linux, we can use renameat2 to do this atomically.
             let _ = unlink(old_path.as_ptr() as *const c_char);
         }
         Ok(())
     }
 }
+
+#[cfg(target_os = "redox")]
+pub fn persist(old_path: &Path, new_path: &Path, overwrite: bool) -> io::Result<()> {
+    // XXX implement in better way when possible
+    if !overwrite && new_path.exists() {
```

Comment:
I don't think so; this can be fixed once that is possible.

It's actually somewhat worse; `rename` is not atomic either currently (just copy and unlink); that really should be implemented, but it hasn't been done yet.